lib/ensime.coffee

```diff
@@ -374,6 +379,34 @@ module.exports = Ensime =
   consumeLinter: (@indieLinterRegistry) ->
 
 
+  provideIntentions: ->
+    getIntentions = (req) =>
+      textEditor = req.textEditor
+      bufferPosition = req.bufferPosition
+      
+      new Promise (resolve) =>
+        @importSuggestions.getImportSuggestions(
+          @clientOfEditor(textEditor),
+          textEditor.getBuffer(),
+          textEditor.getBuffer().characterIndexForPosition(bufferPosition),
+          textEditor.getWordUnderCursor(), # FIXME!
```

Comment:
Good catch, I forgot about that one. Well, it's leaky. We should use the position from intentions call, not from texteditor cursor. Currently I think you can only trigger the intentions command from cursor so it works. But say they added possibility to right click, then cursor could be somewhere else.

* I don't know if ensime-server even uses it, it got the buffer position too, right? Maybe it's used to sanity check if buffer is dirty.
* I couldn't find a atom api for "getWordUnderPosition", but I guess we could read the source a bit and figure it out.